website/docs/concepts/why_immutability.mdx

```diff
@@ -11,6 +11,14 @@ import {
   ConditionalSnippet,
 } from "../../src/components/CodeSnippet";
 
+Riverpod generally recommends leveraging immutability as often as possible. *Mutability* should only be considered in 
```

Comment:
Sounds like we want a "Why use immutability" block. Likely placed after the  "What is" block.
Although this fine line could reasonably stay here.

It'd expand on the ref.listen/select (& ProviderObserver too) issue with a code snippet showcasing the issue.

Reply:
When you say "this line could stay here", do you mean both the first and second sentence?

> Riverpod generally recommends leveraging immutability as often as possible. *Mutability* should only be considered in specific cases for possible performance optimization, after careful benchmarking to make sure it's worth it.

Or do you mean only the first sentence?

> Riverpod generally recommends leveraging immutability as often as possible.

Reply:
I took the liberty of assuming that both the first and second sentence should be at the top. I put the rest in a "Why Immutability" section. Let me know if this is what you were looking for.

Reply:
Can you take a look at the [`why_immutability_examples.dart`](https://github.com/rrousselGit/riverpod/pull/2526/files#diff-f63fde613c878481e0d23b36345ed7bfbb19318d1fba2d8ba6e19ce18274e4b7) file I created to see if this is what you had in mind for code examples?